packages/core/useContextMenu/index.ts

```diff
@@ -0,0 +1,231 @@
+    hideOnClick = true,
+    onVisibleChange = noop,
+    onContextMenu = noop,
+    target,
+  } = options
+
+  const menuElementRef = computed(() => unrefElement(menuElement))
+  const targetRef = computed(() => unrefElement(target))
+
+  const menuElementSize = reactive(
+    useElementSize(
+      menuElementRef,
+      { width: 0, height: 0 },
+      { box: 'border-box' },
+    ),
+  )
+  const windowSize = reactive(
+    useWindowSize({
+      includeScrollbar: false,
+    }),
+  )
+
+  const enabled = ref(true)
+  const visible = ref(false)
+  const position = ref<Position>({ x: 0, y: 0 })
+
+  // see: https://github.com/vueuse/vueuse/pull/2136#issuecomment-1230525648
+  const menuPositionOffset = computed(() => {
+    const { x, y } = position.value
+    let [left, top, right, bottom]: Array<`${number}px` | null> = [
+        `${x}px`,
+        `${y}px`,
+        null,
+        null,
+    ]
+
+    const overflowX = x + menuElementSize.width > windowSize.width
+    const overflowY = y + menuElementSize.height > windowSize.height
+
+    if (overflowX)
+      [left, right] = [null, `${windowSize.width - x}px`]
+
+    if (overflowY)
+      [top, bottom] = [null, `${windowSize.height - y}px`]
+
+    return {
+      left,
+      top,
+      right,
+      bottom,
+    }
+  })
+
+  const show = () => visible.value = true
+  const hide = () => visible.value = false
+
+  // hide first when it changes
+  watch(enabled, hide)
+  watch(visible, onVisibleChange)
+
+  // automatically show/hide and update the position of the `MenuElement`
+  const stopUpdate = watchEffect(
+    () => {
+      const element = menuElementRef.value
+
+      if (!element)
+        return
+
+      element.style.position = 'fixed'
+      element.style.visibility = visible.value ? 'visible' : 'hidden'
+
+      // performance
+      if (visible.value) {
+        for (const [property, offset] of Object.entries(menuPositionOffset.value))
+          element.style.setProperty(property, offset)
+      }
+    })
+
+  const cleanups = [stopUpdate]
+  const stop = () => {
+    enabled.value = false
+    // wait till the `hide` effect is done.
+    nextTick(() => {
+      cleanups.forEach(fn => fn())
+    })
+  }
+  const contextMenuHandler = (e: MouseEvent) => {
+    if (!enabled.value)
+      return
+
+    if (onContextMenu(e) === false)
+      return
+
+    e.preventDefault()
+    position.value = {
+      x: e.clientX,
+      y: e.clientY,
+    }
+    show()
+    // prevent other(parent) context menu from showing
+    e.stopPropagation()
+  }
+
+  // with `target` specified, apply on it.
+  // otherwise, apply on the entire page.
+  if (target)
+    cleanups.push(useEventListener(targetRef, 'contextmenu', contextMenuHandler))
+  else
+    cleanups.push(useEventListener('contextmenu', contextMenuHandler))
+
+  cleanups.push(
+    useEventListener('scroll', hide),
+    useEventListener('click', hide),
+    useEventListener('resize', hide),
+    useEventListener('blur', hide),
```

Comment:
💡 Got an idea about `useEventListener`
```ts
// multi event
useEventListener(['scroll', 'click', 'resize', 'blur'], handler)

// multi handler
useEventListener('scroll' [handler1, handler2, handler3])

// multi event & multi handler
useEventListener(['scroll', 'click', 'resize', 'blur'], [handler1, handler2, handler3])
```

Reply:
seems not to be the right place, I will place this idea somewhere else later on.😂